driver-core/src/test/java/com/datastax/driver/core/policies/ErrorAwarePolicyTest.java

```diff
@@ -0,0 +1,388 @@
+     * <li>On newQueryPlan, {@link Clock#getTime} is called for each Host that is used.  So if the first host
+     * succeeds, it is only called once.  If the first host is ignored and second succeeds, it is called twice and
+     * so on.</li>
+     * </ol>
+     * <p>
+     * Mock details:
+     * <ol>
+     * <li>Each invocation of {@link Clock#getTime()} returns the previous time + timeIncrements.</li>
+     * <li>Each invocation of {@link Clock#getTick()} returns 0.  {@link #advanceTick()} should be used to advance the
+     * clock ticks by 5s1ns.</li>
+     * </ol>
+     * <p>
+     * Based on these details:
+     * <ol>
+     * <li>When session.execute() is called, the 'system' clock time will advance timeIncrements for each host queried
+     * when they are evaluated in the query plan.</li>
+     * <li>When an error occurs that puts the error rate at or above the error threshold, the 'system' clock time
+     * will advance timeIncrements.</li>
+     * <li>Calling advanceTick and then executing a query that returns an error should cause the Meter to tick and
+     * thus update its rates</li>
+     * </ol>
+     * <p>
+     * If any of these implementation details change, this will cause the behavior of the tests to change and likely
+     * fail.
+     *
+     * @param timeIncrements Amount of time to advance per {@link Clock#getTime()} call.
+     * @param timeUnit       The unit to advance time in.
+     * @return a mock clock to behave as documented above.
+     */
+    private static Clock mockClock(long timeIncrements, TimeUnit timeUnit) {
+        Clock clock = mock(Clock.class);
+        final AtomicLong systemTime = new AtomicLong(0L);
+        final long inc = TimeUnit.MILLISECONDS.convert(timeIncrements, timeUnit);
+        when(clock.getTime()).thenAnswer(new Answer<Long>() {
+            @Override
+            public Long answer(InvocationOnMock invocation) throws Throwable {
+                while (true) {
+                    long previousTime = systemTime.get();
+                    long newTime = previousTime + inc;
+                    if (systemTime.compareAndSet(previousTime, previousTime + inc)) {
+                        return newTime;
+                    }
+                }
+            }
+        });
+        // Make the clock 'tick-less' by default unless the user configures it to tick.
+        when(clock.getTick()).thenReturn(0L);
+
+        return clock;
+    }
+
+    private void prime(int node, Result result) {
+        sCluster.node(node).primingClient().prime(
+                queryBuilder()
+                        .withQuery(QueryTracker.QUERY)
+                        .withThen(then().withResult(result))
+                        .build()
+        );
+    }
+
+    private void awaitTrackerUpdate() {
+        // TODO: If possible, find a way to more deterministically ensure the tracker was called.
+        Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
```

Comment:
Unfortunate to sleep here and I suspect that this may cause intermittent test failures.  This is because the call to update the tracker is done a separate thread after the future tied to execute is completed.

One idea I have is to register a LatencyTracker after the LBP is init'd, with the theory that this tracker would be updated after the one in the LBP's tracker.  I could then use that to detect when the tracker is called and allowing us to advance.  Any other ideas?